platform/core/src/services/ViewportGridService/ViewportGridService.ts

```diff
@@ -184,14 +184,27 @@ class ViewportGridService extends PubSubService {
     this.setDisplaySetsForViewports([props]);
   }
 
-  public async setDisplaySetsForViewports(props) {
-    await this.serviceImplementation._setDisplaySetsForViewports(props);
+  public async setDisplaySetsForViewports(viewportsToUpdate) {
+    const { cineService } = this.servicesManager.services;
+    await this.serviceImplementation._setDisplaySetsForViewports(viewportsToUpdate);
     const state = this.getState();
     const updatedViewports = [];
 
     const removedViewportIds = [];
 
-    for (const viewport of props) {
+    // Stopping the cine of modified viewports before changing the viewports to
+    // avoid inconsistent state and lost references
+    viewportsToUpdate.forEach(viewport => {
```

Comment:
This doesn't feel right. 

Using the `cine` service inside `viewportGridService` is tight coupling, which is really not ideal.

Reply:
Yes. I agree it is not ideal. However it must be done BEFORE the update happens. 

I tried adding a listener to the CineService for the `ViewportGridService.EVENTS.GRID_STATE_CHANGED` event, but it doesn't help, since it happens AFTER this function has done its job and because of that trying to turn off the cine does not work at this point.

One alternative would be find every place where this ViewportGridService function is called and also do it there.

But maybe you have a better suggestion than this one.

Reply:
I could create a command that does that and runs the original function and call it in place of the `viewportGridService.setDisplaySetsForViewports`. And substitute everywhere where it is called.

This way only the commandsModule will know about the cineService and the function will not have to know about it.

Reply:
> I could create a command

@sedghi I did it the way I just described above so you can take a look and see if you like it.